src/saltext/vmware/states/datastore.py

```diff
@@ -0,0 +1,88 @@
+# SPDX-License-Identifier: Apache-2.0
+import logging
+
+import saltext.vmware.utils.common as utils_common
+import saltext.vmware.utils.connect as connect
+import saltext.vmware.utils.vm as utils_vm
+from saltext.vmware.modules.tag import update
+
+log = logging.getLogger(__name__)
+
+try:
+    from pyVmomi import vim
+
+    HAS_PYVMOMI = True
+except ImportError:
+    HAS_PYVMOMI = False
+
+
+__virtualname__ = "vmware_datastore"
+__proxyenabled__ = ["vmware_datastore"]
+
+
+def __virtual__():
+    if not HAS_PYVMOMI:
+        return False, "Unable to import pyVmomi module."
+    return __virtualname__
+
+
+def maintenance_mode(name, maintenance_mode, datacenter_name=None, service_instance=None):
```

Comment:
**question** would `set_maintenance_mode` be a better name? Also thinking it the argument could then be either renamed to `mode` or `activate_maintenance_mode` :thinking:

Reply:
Actually, after a cursory glance at https://docs.vmware.com/en/VMware-vSphere/7.0/com.vmware.vsphere.resmgmt.doc/GUID-D229556E-C991-41BD-A378-363CC9D2B2CC.html

I think it makes sense to do:

```suggestion
def maintenance_mode(name, enter_maintenance_mode, datacenter_name=None, service_instance=None):
```

(and update the docstring to match)